FeLib/Source/graphics.cpp

```diff
@@ -292,6 +294,23 @@ void graphics::BlitDBToScreen()
 
 #endif
 
+void graphics::SetScale(int NewScale)
+{
+  Scale = NewScale;
+#if SDL_MAJOR_VERSION == 1
+#warning Graphics scaling not implemented for SDL v1
+#else
+  // Scale the window, maintaining its center position.
+  v2 WindowPos, OldSize, NewSize = Res * NewScale;
+  SDL_GetWindowPosition(Window, &WindowPos.X, &WindowPos.Y);
+  SDL_GetWindowSize(Window, &OldSize.X, &OldSize.Y);
+  WindowPos += (OldSize - NewSize) / 2;
+  SDL_SetWindowPosition(Window, WindowPos.X, WindowPos.Y);
+  SDL_SetWindowSize(Window, Res.X * NewScale, Res.Y * NewScale);
+  SDL_RenderSetScale(Renderer, NewScale, NewScale);
```

Comment:
I noticed this works even without this line. I don't know if this is just platform-specific behavior or an SDL bug so I'll leave it.